src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,140 @@
+            int doneIndex = Integer.parseInt(fullCommand[1]);
+            checkIndex(doneIndex, tasks.size());
+            return new DoneCommand("done", doneIndex);
+        case "delete":
+            if (fullCommand.length == 1) {
+                throw new DukeException("Please give an index number");
+            }
+            int deleteIndex = Integer.parseInt(fullCommand[1]);
+            checkIndex(deleteIndex, tasks.size());
+            return new DeleteCommand("delete", deleteIndex);
+        case "todo":
+            desc = commandLine.replace("todo", "").trim();
+            if (desc.equals("")) {
+                throw new DukeException("A to-do needs a description");
+            }
+            return new AddTodoCommand(desc);
+        case "deadline":
+            String by;
+
+            String[] deadlineArgs = commandLine.replace("deadline ", "").split("/by ");
+
+            if (deadlineArgs.length != 2) {
+                throw new DukeException("Invalid format for deadline");
+            }
+
+            desc = deadlineArgs[0];
+            by = deadlineArgs[1];
+
+            LocalDateTime deadlineDate = this.parseDate(by);
+            return new AddDeadlineCommand(desc, deadlineDate);
+
+        case "event":
+            String at;
+
+            String[] eventArgs = commandLine.replace("event ", "").split("/at ");
+
+            if (eventArgs.length != 2) {
+                throw new DukeException("Invalid format for event");
+            }
+
+            desc = eventArgs[0];
+            at = eventArgs[1];
+
+            if (desc.equals("")) {
+                throw new DukeException("An event needs a description");
+            }
+
+            LocalDateTime eventDate = this.parseDate(at);
+            return new AddEventCommand(desc, eventDate);
+
+        case "find":
+            StringBuilder keywordBuilder = new StringBuilder();
+            for (int i = 1; i < fullCommand.length; i++) {
+                if (i != 1) {
+                    keywordBuilder.append(" ");
+                }
+                keywordBuilder.append(fullCommand[i]);
+            }
+            return new FindCommand("find", keywordBuilder.toString());
+
+        default:
+            throw new DukeException("I do not understand that command");
+        }
+    }
+
+    /**
+     * Checks if the index number that the user input is valid.
+     *
+     * @param i            the index number that the user input.
+     * @param lengthOfList the length of the user's TaskList.
+     * @throws DukeException If the index number is negative or more than length of the TaskList.
+     */
+    public static void checkIndex(int i, int lengthOfList) throws DukeException {
+        if (i <= 0) {
+            throw new DukeException("Please give an index number > 0");
+        } else if (i > lengthOfList) {
+            throw new DukeException("Maximum index number is " + lengthOfList);
+        }
+    }
+
+    /**
+     * Parses the date
```

Comment:
JavaDoc comments can be more descriptive, e.g. "Parses the date string given by the user into a LocalDateTime object."